lib/helpers/redflags.js

```diff
@@ -0,0 +1,46 @@
+const markRedFlag = (s) => {
+  let redflag = "";
+
+  //Email validation
+
+  if (!validateEmail(s.email)) {
+    redflag += "emial is not valid,";
+  }
+  //Job kab lagega validation
+  //if this is empty or date is refering to the past
+  const currDate = new Date();
+  const jDate = new Date(s.jobKabLagega);
+  if (
+    s.jobKabLagega == null ||
+    (jDate.getDate() < currDate.getDate() &&
+      jDate.getMonth() < currDate.getMonth())
+  ) {
+    redflag += "Job kab lagegi is not valid,";
+  }
+
+  //If evaluation field is empty
+  if (s.evaluation == null) {
+    redflag += "evaluation is not available,";
+  }
+  return redflag;
+};
+const joinDateValidation = (transitions) => {
+  let count = 0;
+  transitions.map((transition) => {
+    if (transition.to_stage === "finallyJoined") {
+      count += 1;
+    }
+  });
+  if (count > 1) {
+    return false;
+  }
+  return true;
```

Comment:
Ideally, the purpose of the functional programming construct `map` is to create a new `Array` of outputs from an existing `Array` of inputs without modifying existing state such as a `counter` variable (e.g., `const doubledArray = arr.map(x => 2*x)`).  Here, since the callback function to `map` doesn't return anything, you're unnecessarily creating a new Array of `undefined` that's not being stored.

`forEach` is more appropriate for an imperative approach when you're giving commands such as: Increment `count` in the case that you find a transition whose `to_stage` property is `finallyJoined`.  But a simple way for counting the number of items in a list that meet a certain condition is to use `filter` to create a new Array only containing the items for which the condition is `true` and then getting its `length`:
```suggestion
  return transitions.filter((transition) => transition.to_stage === 'finallyJoined')
      .length <= 1;
```
Note a couple of potential drawbacks with this code:
1. You're creating a new `Array` when all you want is the `count`.  This is probably not a big deal here because the number of items whose `to_state` is `"finallyJoined"` is probably small, but an alternative here that won't create a new list is the following: 
```
transitions.reduce((currentCount, transition) => transition.to_stage === 'finallyJoined' ?
    currentCount + 1 : currentCount, 0);
```
2. You only want to check if the count is 0 or 1 so if you ever see that the count is 2, you should stop going through the Array.  An alternative that does this would be the following:
```
let count = 0;
return transitions.every((transition) => transition.to_stage !== 'finallyJoined' || ++count < 2);
```
This works because of short-circuit evaluation, mainly every time `transition.to_stage !== 'finallyJoined'`, it doesn't need to check what comes after `||` because inclusive or is always `true` whenever one of the conditions is.  In the case that it's `false`, the `count` will be incremented and checked to make sure that it's less than 2.  If not, then the entire condition is `false` so `transitions.every` will immediately return `false` (because the condition is not `true` for every `transition`).  Otherwise if it gets through the entire Array with the condition always being `true`, then `true` will be returned.

Reply:
Thanks for feedback .